### ISSUE TITLE
Blacklisting seems to be broken? 

### ISSUE THREAD

**Issue by magicmiketm** [zlainsama/PeacefulSurface#25]
Here is my config

```
{
  "Disabled": false,
  "Living": false,
  "Mob": true,
  "Animal": false,
  "Tameable": false,
  "Checking_LightLevel": true,
  "Sunny": false,
  "Raining": false,
  "Thundering": true,
  "Day": false,
  "Night": false,
  "InvertedMobFilter": true,
  "InvertedDimensionFilter": false,
  "InvertedLightLevelChecking": false,
  "mobFilter": "(\\bZombie\\b)|(\\bSkeleton\\b)|(\\bCreeper\\b)|(\\bSpider\\b)|(\\bHusk\\b)|(\\bSkeleton_Warrior\\b)|(\\bStray\\b)|(\\bFestive_Creeper\\b)|(\\bSupport_Creeper\\b)|(\\bRocket_Creeper\\b)|(\\bAngry_Zombie\\b)",
  "dimensionFilter": "(\\bthe_end\\b)|(\\bthe_nether\\b)",
  "LightLevel": 0,
  "MoonPhase": 1,
  "DisabledUnderBloodmoon": false,
  "DisabledWhenSunny": false,
  "DisabledWhenRaining": false,
  "DisabledWhenThundering": false,
  "DisabledWhenDay": false,
  "DisabledWhenNight": false
}
```
Essentially, with InvertedMobFilter set to true and mobs written into the mobFilter, it's supposed to blacklist the mobs written into the filter. But only the mobs that only have one word names such as Creeper for example are managing to be blacklisted, mobs that have names with spaces and such are not being recognized. They still manage to spawn on the surface (circled are the blacklisted mobs): 
![image](https://user-images.githubusercontent.com/39220604/132962275-9ecd25fc-34e8-4f1d-9d97-06400589a010.png)

**Comment by zlainsama** [comment]
Which version are you using?
Depend on the version, you might need to configure mobFilter differently.

**Comment by magicmiketm** [comment]
I am using the 1.12.2. version of the mod.

**Comment by magicmiketm** [comment]
https://www.curseforge.com/minecraft/mc-mods/peacefulsurface/files/2727271

**Comment by zlainsama** [comment]
What mod are those mobs coming from?

I think there are two possible causes for this issue:
1. The names specified in mobFilter do not match with the actual names used in that mod.
2. They are spawned differently from normal mobs, bypassing the CheckSpawn event that my mod relies on to filter mobs.

**Comment by magicmiketm** [comment]
These mobs are from the mod Primitive Mobs an Thaumcraft, they spawn just like any other hostile mob such as the Zombie or the Creeper. I tried a variation where the space between their names is an actual white space and a underscore and neither of these work, it's probably the 1st option. The second option however cannot be as the mod worked as intended when I didn't have the blacklist (AKA. "InvertedMobFilter" wasn't to true) enabled and creatures from Primitive Mobs did not spawn on the surface.

**Comment by zlainsama** [comment]
The AngryZombie is referred as "BrainyZombie" in Thaumcraft' code.
Those creepers and skeleton warrior is referred as "rocket_creeper", "festive_creeper", "support_creeper", "skeleton_warrior" in [PrimitiveMobs' code](https://github.com/Daveyx0/PrimitiveMobs/blob/master/src/main/java/net/daveyx0/primitivemobs/core/PrimitiveMobsEntityRegistry.java).

**Comment by magicmiketm** [comment]
I see, so your config's filters are literally checking for the mobs' IDs and not the mob's display names. Correct?

**Comment by zlainsama** [comment]
Yes, in old versions, those ids are used when spawning those entities.

In newer versions, you will need formal registry ids such as "minecraft:slime".

**Comment by magicmiketm** [comment]
No matter what I tried, blacklisting still does not work regardless of how I formatted the syntax. Here is the most recent code I tried using to detect a mob. 
```
{
  "Disabled": false,
  "Living": false,
  "Mob": true,
  "Animal": false,
  "Tameable": false,
  "Checking_LightLevel": true,
  "Sunny": false,
  "Raining": false,
  "Thundering": false,
  "Day": false,
  "Night": false,
  "InvertedMobFilter": true,
  "InvertedDimensionFilter": false,
  "InvertedLightLevelChecking": false,
  "mobFilter": "(([a-z]{0,}spider)|([a-z]{0,}.spider))|(([a-z]{0,}creeper)|([a-z]{0,}.creeper))|(([a-z]{0,}zombie)|([a-z]{0,}.zombie))|(([a-z]{0,}skeleton)|([a-z]{0,}.skeleton))",
  "dimensionFilter": "(\\bthe_end\\b)|(\\bthe_nether\\b)",
  "LightLevel": 0,
  "MoonPhase": 0,
  "DisabledUnderBloodmoon": false,
  "DisabledWhenSunny": false,
  "DisabledWhenRaining": false,
  "DisabledWhenThundering": false,
  "DisabledWhenDay": false,
  "DisabledWhenNight": false
}
```
I used the javascript flavour variant of Regular Expressions' syntax as an example to build up the code. Do you have any resources I could read that can teach me what type of Regex syntax you're using so I can configure it properly?

Which one of these strings does the config actually look for in the 1.12.2. v2 update of your mod? 
![image](https://user-images.githubusercontent.com/39220604/133466920-9bd9ee64-7ca9-4ce6-b000-69c4f32d03a1.png)

**Comment by magicmiketm** [comment]
Which one of these things does your Regex actually look for? I can't figure it out.
![image](https://user-images.githubusercontent.com/39220604/133467689-9b3471b4-e83b-4414-95cd-71782651fac3.png)

Here's the code I've managed to construct from the javascript flavour of Regular Expressions' syntax.
```
{
  "Disabled": false,
  "Living": false,
  "Mob": true,
  "Animal": false,
  "Tameable": false,
  "Checking_LightLevel": true,
  "Sunny": false,
  "Raining": false,
  "Thundering": false,
  "Day": false,
  "Night": false,
  "InvertedMobFilter": true,
  "InvertedDimensionFilter": false,
  "InvertedLightLevelChecking": false,
  "mobFilter": "(([a-z]{0,}spider)|([a-z]{0,}.spider))|(([a-z]{0,}creeper)|([a-z]{0,}.creeper))|(([a-z]{0,}zombie)|([a-z]{0,}.zombie))|(([a-z]{0,}skeleton)|([a-z]{0,}.skeleton))",
  "dimensionFilter": "(\\bthe_end\\b)|(\\bthe_nether\\b)",
  "LightLevel": 0,
  "MoonPhase": 0,
  "DisabledUnderBloodmoon": false,
  "DisabledWhenSunny": false,
  "DisabledWhenRaining": false,
  "DisabledWhenThundering": false,
  "DisabledWhenDay": false,
  "DisabledWhenNight": false
}
```

**Comment by zlainsama** [comment]
In 1.12.2, the string my regex looking for is just some hardcoded magic string, there is no regulation in that time, modders can choose whatever they want, you need to look into their code.
Vanilla use something like 'Creeper', 'Zombie'.
I think you just need to use '\\' instead of single '\' in normal regex.
